src/OpenLoco/src/Ui/Windows/Vehicle.cpp

```diff
@@ -2878,22 +2880,256 @@ namespace OpenLoco::Ui::Windows::Vehicle
             return args;
         }
 
+        // 0x004B5BB9
+        static ViewportInteraction::InteractionArg stationLabelAdjustedInteraction(const Vehicles::VehicleHead& head, int16_t var_842, StationId stationId, ViewportInteraction::InteractionArg interaction)
+        {
+            auto* station = StationManager::get(stationId);
+            if (station == nullptr)
+            {
+                return ViewportInteraction::kNoInteractionArg;
+            }
+            if (station->owner != CompanyManager::getControllingId())
+            {
+                return ViewportInteraction::kNoInteractionArg;
+            }
+
+            _hoveredStationId = stationId;
+            World::setMapSelectionFlags(World::MapSelectionFlags::hoveringOverStation);
+            ViewportManager::invalidate(station);
+
+            {
+                auto args = FormatArguments::mapToolTip(StringIds::click_to_insert_new_order_stop_at);
```

Comment:
does this scoped block do anything? what is the end result?

Reply:
FormatArguments::mapToolTip is a nice weird global var this is putting the values into the global. I scoped it so that I don't reuse the variable in the code later on as I want it to overwrite the global variable. Eventually all of this would be changed to:
```
auto args = FormatArguments(StringIds::...);
args.push(...);
setMapTooltipMessage(args);
```
but that isn't how it works yet.